### ISSUE TITLE
Add option to autocomplete while maintaining case first letter

### ISSUE THREAD

**Issue by JoshKasap** [tth05/obsidian-completr#10]
If I start typing a word and then use completr to finish that word it overrides the case of the first letter. In most cases I would prefer to maintain the case of the original word I'm typing. 
![image](https://user-images.githubusercontent.com/56367532/154373052-a1d43876-d99d-4900-a58f-5e8b12dbb6cb.png)

This is also a problem in that if both upper and lowercase versions of a word are stored in completr and you start typing out the uppercase version the lowercase version will show above it. When both are shown I would expect the one that closest matches the case of the typed word to be displayed first. 
![image](https://user-images.githubusercontent.com/56367532/154373799-c811f766-43fc-4600-82f3-649f8dd364d0.png)
Of course, when this is the case it might just be a better idea to merge the displayed options into one and display the suggestion as a single entry in the case that matches what the user typed. (but that's up to you)

I'm sure some people might prefer completr to override text case as it does now so having this be an option that can be changed in the settings would be ideal. However, I would argue the default behavior should be to maintain the case the user provided if possible.

**Comment by tth05** [comment]
That looks like you want this:
![image](https://user-images.githubusercontent.com/36999320/154474287-7e8e84bc-78b3-4358-afd1-9fcbb69ca8a1.png)

**Comment by tth05** [comment]
Seems like it broke when I did some refactoring. Should be working now!
Fixed in 18376d38667f977fedd56611b565aa57df581ce8

**Comment by tth05** [comment]
@JoshKasap Does this work as expected now?

**Comment by MatteCrystal** [comment]
Almost but just slightly off. It will match the case if that matching case was stored. Ex: I stored the name Molly so if I start typing it with a capital `M` I will get the suggestion. 
![image](https://user-images.githubusercontent.com/32424764/155456225-8aa3237f-bad3-43c9-944c-cf8f8439b798.png)

But if I start typing molly with a lower case `m` I don't get any suggestions because I've never written molly in all lower case. 
![image](https://user-images.githubusercontent.com/32424764/155456440-d547d28c-a9db-4317-b535-bd80fd6fe225.png)

Ideally, I would expect to get the suggestion of molly in both scenarios and the suggestion case would match whether I started with upper or lower case. 

So it should match case on the user input side but ignore the case of stored words if that makes sense. 

Another example: let's say I had `itunes` as a stored word and nothing else. Then I typed `iTun`, the autocomplete suggestion would match what I typed so far to the stored word `itunes`. But the displayed autocomplete suggestion would not be `itunes`. It would be `iTunes` because it matches the case format I already typed and then uses the rest of the stored word to autocomplete the last portion.

**Comment by tth05** [comment]
Now that just sounds like exactly what `Ignore-Case & Append` does, as I mentioned in my [first reply](https://github.com/tth05/obsidian-completr/issues/10#issuecomment-1042861613).

**Comment by MatteCrystal** [comment]
ahh ok I was using `match case and replace` It's a little confusing understanding what each one does. 

Anyway, I swapped to `Ignore-Case & Append` It's basically what I described except for one small issue. It displays the result multiple times. 

![image](https://user-images.githubusercontent.com/32424764/155563491-9da84131-84f7-406f-82eb-5dd0a923e06f.png)

So i have iTunes stored in 3 different ways with different case. `Ignore-Case & Append` properly maintains the case of what I typed and displays the autocomplete suggestion correctly. But Ideally, it would be able to recognize that the other suggestions are the same and only display one.
